Build failure: tenacity

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build nixpkgs/946738808f0ddc9e57ee0df76aa5986715c24676#tenacity`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318195043

### Relevant log output

```console
FAILED: [code=1] libraries/lib-utility/CMakeFiles/lib-utility.dir/MemoryX.cpp.o 
/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/g++ -DEXPERIMENTAL_ -DEXPERIMENTAL_DRAGGABLE_PLAY_HEAD -DEXPERIMENTAL_EASY_CHANGE_KEY_BINDINGS -DEXPERIMENTAL_EFFECT_MANAGEMENT -DEXPERIMENTAL_FULL_WASAPI -DEXPERIMENTAL_HALF_WAVE -DEXPERIMENTAL_IMPORT_AUP3 -DEXPERIMENTAL_KEY_VIEW -DEXPERIMENTAL_MODULE_PREFS -DEXPERIMENTAL_NOISE_REDUCTION -DEXPERIMENTAL_NOTETRACK_OVERLAY -DEXPERIMENTAL_NYQUIST_SPLIT_CONTROL -DEXPERIMENTAL_PUNCH_AND_ROLL -DEXPERIMENTAL_REALTIME_AUDACITY_EFFECTS -DEXPERIMENTAL_SCIENCE_FILTERS -DEXPERIMENTAL_SCROLLING_LIMITS -DEXPERIMENTAL_SCRUBBING_SCROLL_WHEEL -DEXPERIMENTAL_SCRUBBING_SUPPORT -DEXPERIMENTAL_SPECTRAL_EDITING -DEXPERIMENTAL_SYNC_LOCK -DEXPERIMENTAL_THEME_PREFS -DEXPERIMENTAL_THEMING -DEXPERIMENTAL_TWO_TONE_TIME_RULER -DEXPERIMENTAL_ZERO_PADDED_SPECTROGRAMS -DEXPERIMENTAL_ZOOM_TOGGLE_BUTTON -DUTILITY_API="" -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -Dlib_utility_EXPORTS -I/build/source/libraries/lib-utility -isystem /nix/store/n0dplq2i6l8mvignksvahzzzz83mk78y-wxwidgets-3.2.8.1/lib/wx/include/gtk3-unicode-3.2 -isystem /nix/store/n0dplq2i6l8mvignksvahzzzz83mk78y-wxwidgets-3.2.8.1/include/wx-3.2 -pthread -O3 -DNDEBUG -std=gnu++17 -fPIC -include /build/source/build/src/private/configunix.h -DTENACITY_VERSION=1 -DTENACITY_RELEASE=3 -DTENACITY_REVISION=4 -DTENACITY_MODLEVEL=0 -DTENACITY_VERSION_STRING=L\"1.3.4\" -DTENACITY_FILE_VERSION=L\"1,3,4,0\" -Dsafenew=new -DWXINTL_NO_GETTEXT_MACRO -U_DEBUG -DIS_RELEASE -MD -MT libraries/lib-utility/CMakeFiles/lib-utility.dir/MemoryX.cpp.o -MF libraries/lib-utility/CMakeFiles/lib-utility.dir/MemoryX.cpp.o.d -o libraries/lib-utility/CMakeFiles/lib-utility.dir/MemoryX.cpp.o -c /build/source/libraries/lib-utility/MemoryX.cpp
/build/source/libraries/lib-utility/MemoryX.cpp: In static member function 'static void* NonInterferingBase::operator new(std::size_t, std::align_val_t)':
/build/source/libraries/lib-utility/MemoryX.cpp:34:36: error: 'uintptr_t' does not name a type
   34 |    auto integer = reinterpret_cast<uintptr_t>(ptr);
      |                                    ^~~~~~~~~
/build/source/libraries/lib-utility/MemoryX.cpp:12:1: note: 'uintptr_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
   11 | #include "MemoryX.h"
  +++ |+#include <cstdint>
   12 |
```

### Additional context

_No response_

https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.1-zen1, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/gep/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/gep/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/gep/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/gep/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/sa7kxynphvibmp8z8dar31j3dyn3nvfj-lix-2.93.3/share`
 - nixpkgs: `/nix/store/1ny7brxnqbx6xilj7mjdlinzpb5a1s3i-source`

### Notify maintainers

@IreneKnapp 
@niklaskorz 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/te/tenacity/cstdlib.patch
+++ b/pkgs/by-name/te/tenacity/cstdlib.patch
@@ -1,0 +1,26 @@
+From f523f6a4272910417da6d8f02597a560a9c594e4 Mon Sep 17 00:00:00 2001
+From: Ethan Carter Edwards <ethan@ethancedwards.com>
+Date: Wed, 31 Dec 2025 21:17:39 -0500
+Subject: [PATCH] fix compilation for gcc 15
+
+Link: https://github.com/NixOS/nixpkgs/issues/475574
+
+Signed-off-by: Ethan Carter Edwards <ethan@ethancedwards.com>
+---
+ libraries/lib-utility/MemoryX.h | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/libraries/lib-utility/MemoryX.h b/libraries/lib-utility/MemoryX.h
+index 2c96af17b..58bf67127 100644
+--- a/libraries/lib-utility/MemoryX.h
++++ b/libraries/lib-utility/MemoryX.h
+@@ -5,6 +5,7 @@
+ #include <iterator>
+ #include <memory>
+ #include <new> // align_val_t and hardware_destructive_interference_size
+ #include <cstdlib> // Needed for free.
++#include <cstdint>
+ #ifndef safenew
+ #define safenew new
+-- 
+2.47.3

--- a/pkgs/by-name/te/tenacity/package.nix
+++ b/pkgs/by-name/te/tenacity/package.nix
@@ -61,6 +61,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-2gndOwgEJK2zDSbjcZigbhEpGv301/ygrf+EQhKp8PI=";
   };
 
+  # https://codeberg.org/tenacityteam/tenacity/pulls/696
+  # can be removed at next version bump
+  patches = [
+    ./cstdlib.patch
+  ];
+
   postPatch = ''
     # GIT_DESCRIBE is used for the version string and can't be passed in
     # as an option, so we substitute it instead.

